website/pages/docs/configuration/service-registration/kubernetes.mdx

```diff
@@ -0,0 +1,155 @@
+---
+layout: docs
+page_title: Kubernetes - Service Registration - Configuration
+sidebar_title: Kubernetes
+description: >-
+  Kubernetes Service Registration labels Vault pods with their current status
+  for use with selectors.
+---
+
+# Kubernetes Service Registration
+
+Kubernetes Service Registration tags Vault pods with their current status for
+use with selectors. Service registration is only available when Vault is running in
+[High Availability mode](/docs/concepts/ha).
+
+- **HashiCorp Supported** – Kubernetes Service Registration is officially supported
+  by HashiCorp.
+## Configuration
+
+```hcl
+service_registration "kubernetes" {
+  namespace      = "my-namespace"
+  pod_name       = "my-pod-name"
+}
+```
+
+Alternatively, the namespace and pod name can be set through the following
+environment variables:
+
+- `VAULT_K8S_NAMESPACE`
+- `VAULT_K8S_POD_NAME`
+
+This allows you to set these parameters using
+[the Downward API](https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/).
+
+If using only environment variables, the service registration stanza declaring
+you're using Kubernetes must still exist to indicate your intentions:
+
+```
+service_registration "kubernetes" {}
+```
+
+## Examples
+
+Once properly configured, enabling service registration will cause Kubernetes pods
+to come up with the following labels:
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: vault
+  labels:
+    vault-active: "false"
+    vault-initialized: "true"
+    vault-perf-standby: "false"
+    vault-sealed: "false"
+    vault-version: 1.3.0
+```
+
+After shutdowns, Vault pods will bear the following labels:
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: vault
+  labels:
+    vault-active: "false"
+    vault-initialized: "false"
+    vault-perf-standby: "false"
+    vault-sealed: "true"
+    vault-version: 1.3.0
+```
+
+## Label Definitions
+
+- `vault-active` `(string: "true"/"false")` – Vault active is updated dynamically each time Vault's active status changes.
+  **This field is only valuable if the pod is NOT a member of a performance standby cluster**,
```

Comment:
The phrasing "is only valuable" is a little vague to me. I find it confusing that this is not applicable to perf-standby setups, since they inherently have an active/standby dynamic as well. e.g. https://www.vaultproject.io/docs/enterprise/performance-standby/#monitoring-performance-standbys shows both the "HA Mode" and "Performance Standby Node" status, which is what I thought these parameters would correspond to?

Reply:
Ah, makes sense, thanks!

Reply:
https://github.com/hashicorp/vault/pull/8374/commits/1156f336f3f2ecce401f48b61abe942adf5f32c8